freedv does not build on darwin

### Describe the bug

Building freedv on darwin fails.

### Steps To Reproduce
Steps to reproduce the behavior:

`nix-build -A freedv` 

### Expected behavior

It builds.

### Additional context

This is the error message from the build:

```
[ 97%] Building CXX object src/CMakeFiles/FreeDV.dir/osx_interface.mm.o
/tmp/nix-build-freedv-1.7.0.drv-0/source/src/osx_interface.mm:23:9: fatal error: 'AVFoundation/AVFoundation.h' file not found
#import <AVFoundation/AVFoundation.h>
        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Copying OS X content src/FreeDV.app/Contents/Resources/freedv.icns
1 error generated.
make[2]: *** [src/CMakeFiles/FreeDV.dir/build.make:556: src/CMakeFiles/FreeDV.dir/osx_interface.mm.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:160: src/CMakeFiles/FreeDV.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
error: builder for '/nix/store/p0zhhfw1n5fxhkzq3zxd91yr0r8ygkm5-freedv-1.7.0.drv' failed with exit code 2
```

### Notify maintainers

Seems the maintainer doesn't exist or `mvs` is not a github name.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.3.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.4`
 - channels(nknezevi1): `""`
 - nixpkgs: `/nix/store/f1s4icgz5zdvwgpc8kb9nr7ab46ak5wv-nixpkgs-src`
```

--- a/pkgs/applications/radio/freedv/default.nix
+++ b/pkgs/applications/radio/freedv/default.nix
@@ -13,6 +13,10 @@
 , hamlib
 , wxGTK31-gtk3
 , pulseSupport ? config.pulseaudio or stdenv.isLinux
+, AppKit
+, AVFoundation
+, Cocoa
+, CoreMedia
 }:
 
 stdenv.mkDerivation rec {
@@ -26,7 +30,15 @@ stdenv.mkDerivation rec {
     hash = "sha256-LPCY5gPinxJkfPfumKggI/JQorcW+Qw/ZAP6XQmPkeA=";
   };
 
+  postPatch = lib.optionalString stdenv.isDarwin ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace "if(APPLE)" "if(0)" \
+      --replace "\''${FREEDV_LINK_LIBS})" "\''${FREEDV_LINK_LIBS} \''${FREEDV_LINK_LIBS_OSX})" \
+      --replace "\''${RES_FILES})" "\''${RES_FILES} \''${FREEDV_SOURCES_OSX})"
+  '';
+
   nativeBuildInputs = [ cmake ];
+
   buildInputs = [
     codec2
     libsamplerate
@@ -35,7 +47,13 @@ stdenv.mkDerivation rec {
     speexdsp
     hamlib
     wxGTK31-gtk3
-  ] ++ (if pulseSupport then [ libpulseaudio ] else [ portaudio ]);
+  ] ++ (if pulseSupport then [ libpulseaudio ] else [ portaudio ])
+  ++ lib.optionals stdenv.isDarwin [
+    AppKit
+    AVFoundation
+    Cocoa
+    CoreMedia
+  ];
 
   cmakeFlags = [
     "-DUSE_INTERNAL_CODEC2:BOOL=FALSE"
@@ -46,8 +64,7 @@ stdenv.mkDerivation rec {
     homepage = "https://freedv.org/";
     description = "Digital voice for HF radio";
     license = licenses.lgpl21;
-    maintainers = with maintainers; [ mvs ];
+    maintainers = with maintainers; [ mvs wegank ];
     platforms = platforms.unix;
-    broken = stdenv.isDarwin;  # see https://github.com/NixOS/nixpkgs/issues/165422
   };
 }

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -27671,7 +27671,9 @@ with pkgs;
       shiboken2;
   };
 
-  freedv = callPackage ../applications/radio/freedv { };
+  freedv = callPackage ../applications/radio/freedv {
+    inherit (darwin.apple_sdk.frameworks) AppKit AVFoundation Cocoa CoreMedia;
+  };
 
   freemind = callPackage ../applications/misc/freemind {
     jdk = jdk8; # TODO: remove override https://github.com/NixOS/nixpkgs/pull/89731

